ui/basic-auth.js

```diff
@@ -0,0 +1,18 @@
+/* Exports a single function, which installs a basic auth check if that's how
+ * we're configured. Uses a constant-time buffer comparison */
+import bufferEq from 'buffer-equal-constant-time';
+import httpAuth from 'http-auth';
+
+export default function (envCreds, expressApp) {
```

Comment:
Is this a standard pattern for doing the check? I ask only because I don't feel qualified to truly review this piece. I have spent time wrapping my head around it enough to know that I wouldn't be able to see any nuanced issues. If it's the way everyone does this sort of thing, then I'm happy to rely on the knowledge of many. If you feel very comfortable with this code, @cmc333333, that's also good enough for me.